django-stubs/db/models/fields/__init__.pyi

```diff
@@ -205,10 +205,10 @@ class Field(RegisterLookupMixin, Generic[_ST, _GT]):
     # TODO: plugin support
     def formfield(
         self,
-        form_class: type[Field] | None = ...,
+        form_class: type[forms.Field] | None = ...,
         choices_form_class: type[forms.ChoiceField] | None = ...,
         **kwargs: Any,
```

Comment:
If you're feeling adventurous, it would be useful to typehint additional keys supported in `kwargs`, in a separate PR. Looking at the `formfield()` method code, seems like you can copy certain arguments from `CharField` constructor:

```python
    def formfield(self, form_class=None, choices_form_class=None, **kwargs):
            ...
            for k in list(kwargs):
                if k not in (
                    "coerce",
                    "empty_value",
                    "choices",
                    "required",
                    "widget",
                    "label",
                    "initial",
                    "help_text",
                    "error_messages",
                    "show_hidden_initial",
                    "disabled",
                ):
                    del kwargs[k]
        defaults.update(kwargs)
        if form_class is None:
            form_class = forms.CharField
        return form_class(**defaults)
```